index.js

```diff
@@ -17,50 +17,59 @@ const normalizeEmitter = emitter => {
 	};
 };
 
-module.exports = (emitter, event, options) => {
-	let cancel;
-
-	const ret = new Promise((resolve, reject) => {
-		if (typeof options === 'function') {
-			options = {filter: options};
-		}
+const multiple = (emitter, event, options) => {
+	options = Object.assign({
+		rejectionEvents: ['error'],
+		multiArgs: false,
+		resolveImmediately: false
+	}, options);
 
-		options = Object.assign({
-			rejectionEvents: ['error'],
-			multiArgs: false
-		}, options);
+	if (isNaN(options.count) || options.count < 1) {
+		throw new TypeError('`count` option should be a number greater than 1');
```

Comment:
I don't see the point of not allowing 0 and 1 as count. That just makes it harder to use this API programmatically with unexpected `count` numbers.

```suggestion
		throw new TypeError('The `count` option should be a positive integer or Infinity');
```

The `count` option should also be documented for the above, that it accepts a positive integer or Infinity.

Reply:
Also, why have you removed `the` article from the test name?

Reply:
Ok, so `count >= 0` it is.

Reply:
> Also, why have you removed the article from the test name?

I felt it was moot.

Reply:
Ah, just a shortcut? :P